MovieQuiz/Presentation/MovieQuizViewController.swift

```diff
@@ -1,72 +1,255 @@
 
+        let action = UIAlertAction(title: "Сыграть ещё раз", style: .default) { _ in
+            self.movieQuiz.reset()
+            self.updateView()
+        }
 
- Картинка: Kill Bill
- Настоящий рейтинг: 8,1
- Вопрос: Рейтинг этого фильма больше чем 6?
- Ответ: ДА
+        alert.addAction(action)
 
+        self.present(alert, animated: true, completion: nil)
+    }
 
- Картинка: The Avengers
- Настоящий рейтинг: 8
- Вопрос: Рейтинг этого фильма больше чем 6?
- Ответ: ДА
+    private func updateImageBorder(isPositive: Bool) {
+        let color = isPositive ? UIColor.ypGreen.cgColor : UIColor.ypRed.cgColor
+        imageView.layer.borderColor = color
+        imageView.layer.borderWidth = 8
+    }
+    
+    private func performResultButtonClick(with answer: Answer) {
+        let result = movieQuiz.checkAnswer(answer)
+        updateImageBorder(isPositive: result)
+        noButton.isEnabled = false
+        yesButton.isEnabled = false
+        
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) {
+            self.updateView()
+        }
+    }
+    
+    @IBOutlet private weak var questionLabel: UILabel!
+    @IBOutlet private weak var counterLabel: UILabel!
+    @IBOutlet private weak var imageView: UIImageView!
+    @IBOutlet private weak var textLabel: UILabel!
+    @IBOutlet private weak var noButton: UIButton!
+    @IBOutlet private weak var yesButton: UIButton!
+
+    @IBAction private func noButtonClicked() {
+        performResultButtonClick(with: .no)
+    }
+    
+    @IBAction private func yesButtonClicked() {
+        performResultButtonClick(with: .yes)
+    }
+}
 
+// MARK: - MovieQuizModel
 
- Картинка: Deadpool
- Настоящий рейтинг: 8
- Вопрос: Рейтинг этого фильма больше чем 6?
- Ответ: ДА
+struct MovieQuizModel {
+    private let riddleGenerator: RiddleGenerator
+    private(set) var movieRiddles: [MovieRiddle]
+    private(set) var correctAnswers: Int
+    private(set) var currentRiddle: MovieRiddle
+    private(set) var gameIsEnded: Bool = false
 
+    init(riddleGenerator: RiddleGenerator) {
+        self.riddleGenerator = riddleGenerator
+        movieRiddles = riddleGenerator.generate()
+        correctAnswers = 0
+        currentRiddle = movieRiddles[0]
+    }
 
- Картинка: The Green Knight
- Настоящий рейтинг: 6,6
- Вопрос: Рейтинг этого фильма больше чем 6?
- Ответ: ДА
+    var currentRiddleNum: Int {
+        guard let index = movieRiddles.firstIndex(where: { riddle in riddle.name == currentRiddle.name }) else {
+            return movieRiddles.count
+        }
+        return index + 1
+    }
+    
+    mutating func reset() {
+        gameIsEnded = false
+        movieRiddles = riddleGenerator.generate()
+        correctAnswers = 0
+        currentRiddle = movieRiddles[0]
+    }
+    
+    mutating func checkAnswer(_ answer: Answer) -> Bool {
+        var result = false
+        if currentRiddle.correctAnswer == answer {
+            correctAnswers += 1
+            result = true
+        }
```

Comment:
В другом порядке будет немного читаемее:
```suggestion
        var isResultCorrect = currentRiddle.correctAnswer == answer
        if isResultCorrect {
            correctAnswers += 1
        }
```